server/smart_auth_test.go

```diff
@@ -0,0 +1,75 @@
+package server
+
+import (
+	"bytes"
+	"net/http"
+	"net/http/httptest"
+
+	"github.com/labstack/echo"
+	"github.com/pebbe/util"
+	. "gopkg.in/check.v1"
+)
+
+type SmartAuthSuite struct {
+	Echo *echo.Echo
+}
+
+var _ = Suite(&SmartAuthSuite{})
+
+func (s *SmartAuthSuite) SetUpTest(c *C) {
+	s.Echo = echo.New()
+}
+
+func (s *SmartAuthSuite) TestGetPatientWithoutScopes(c *C) {
+	rr := s.SetUpRequest("GET", "")
+	c.Assert(rr.Code, Equals, http.StatusForbidden)
+}
+
+func (s *SmartAuthSuite) TestGetPatientWithWriteScopes(c *C) {
+	rr := s.SetUpRequest("GET", "user/Patient.write")
+	c.Assert(rr.Code, Equals, http.StatusForbidden)
+}
+
+func (s *SmartAuthSuite) TestGetPatientWithScopes(c *C) {
+	rr := s.SetUpRequest("GET", "user/Patient.read")
+	c.Assert(rr.Code, Equals, http.StatusOK)
+	c.Assert(rr.Body.String(), Equals, "Hello")
+}
+
+func (s *SmartAuthSuite) TestGetPatientWithWildcard(c *C) {
+	rr := s.SetUpRequest("GET", "user/Patient.*")
+	c.Assert(rr.Code, Equals, http.StatusOK)
+	c.Assert(rr.Body.String(), Equals, "Hello")
+}
+
+func (s *SmartAuthSuite) TestGetPatientWithAllWildcard(c *C) {
+	rr := s.SetUpRequest("GET", "user/*.*")
+	c.Assert(rr.Code, Equals, http.StatusOK)
+	c.Assert(rr.Body.String(), Equals, "Hello")
+}
+
+func (s *SmartAuthSuite) TestPostPatientWithScopes(c *C) {
+	rr := s.SetUpRequest("POST", "user/Patient.write")
+	c.Assert(rr.Code, Equals, http.StatusOK)
+	c.Assert(rr.Body.String(), Equals, "Hello")
+}
+
```

Comment:
I was wondering what multiple scopes looks like in the `X-SCOPE` (space separated, comma separated, etc), and hoping to find the answer in a test.  Alas, no.  Perhaps its worth writing a couple of tests with multiple scopes?